pkg/controller/eventing/subscription/reconcile_test.go

```diff
@@ -464,6 +465,33 @@ var testCases = []controllertesting.TestCase{
 				},
 			},
 		},
+	}, {
+		Name: "new subscription to non-existent K8s Service: fails with no service found",
+		InitialState: []runtime.Object{
+			getNewSubscriptionToK8sService(),
+		},
+		WantResult: reconcile.Result{},
+		WantPresent: []runtime.Object{
+			getNewSubscriptionToK8sServiceWithUnknownConditions(),
+		},
+		WantErrMsg: "services \"testk8sservice\" not found",
+		Scheme:     scheme.Scheme,
+		Objects: []runtime.Object{
+			// Source channel
+			&unstructured.Unstructured{
```

Comment:
Can we make this a function?

`getSourceChannelObject()`

Reply:
I like them not wrapped in my opinion. It is easier to move them around and add/modify and not break random tests.

Reply:
It's a tradeoff and maybe it is just 'grass is always greener', but I don't like seeing these ~10 lines repeated in ~10 times. I'll leave it to @vaikas-google to decide. In any case, this should not block the PR.